content/blog/blog-post-10.md

```diff
@@ -0,0 +1,286 @@
+  Kokkos::fence();
+}
+```
+
+This is equivalent to the following python code.
+
+```python
+import numpy as np
+x = np.random.rand(4)
+x_hat = np.fft.rfft(x)
+```
+
+There are two additional arguments in the Kokkos version:
+
+* An instance of the [`ExecutionSpace`](https://kokkos.org/kokkos-core-wiki/API/core/execution_spaces.html)
+* The output View (`x_hat`).
+
+Also, kokkos-fft only accepts [Kokkos Views](https://kokkos.org/kokkos-core-wiki/API/core/view/view.html) as input data. The accessibility of a View from `ExecutionSpace` is statically checked and will result in a compilation error if not accessible.
+
+# Prerequisites
+
+To use kokkos-fft, we need the following:
+
+* `CMake 3.22+`
+* `Kokkos 4.4+`
+
+Depending on your backend, you also need at least one of the following compilers (for the latest requirements, see [README](https://github.com/kokkos/kokkos-fft)):
+
+* `gcc 8.3.0+` (CPUs)
+* `IntelLLVM 2023.0.0+` (CPUs, Intel GPUs)
+* `nvcc 11.0.0+` (NVIDIA GPUs)
+* `rocm 5.3.0+` (AMD GPUs)
+
+# Using kokkos-fft in your CMake project
+
+For the moment, there are two ways to use kokkos-fft: include as a subdirectory in your CMake project or install as a library. Since kokkos-fft is a header-only library, it is easy to add it as a subdirectory in a CMake project. We will see how to do it.
+
+It is assumed that both Kokkos and kokkos-fft are placed under `<project_directory>/tpls`. Here is the structure of a simple CMake project.
+
+```
+---/
+ |
+ └──<project_directory>/
+    |──tpls/
+    |    |──kokkos/
+    |    └──kokkos-fft/
+    |──CMakeLists.txt
+    └──hello.cpp
+```
+
+See [here](https://kokkos.org/kokkos-core-wiki/get-started/integrating-kokkos-into-your-cmake-project.html#embedded-kokkos-via-add-subdirectory-and-git-submodules) about embedding Kokkos via `add_subdirectory()` and Git Submodules in your CMake project. Then, you need to clone the repo of kokkos-fft into `<project_directory>/tpls/kokkos-fft`.
+
+```bash
+git clone --recursive https://github.com/kokkos/kokkos-fft.git
+```
+
+The `CMakeLists.txt` file of the project would be
+
+```CMake
+cmake_minimum_required(VERSION 3.23)
+project(kokkos-fft-as-subdirectory LANGUAGES CXX)
+
+add_subdirectory(tpls/kokkos)
+add_subdirectory(tpls/kokkos-fft)
+
+add_executable(hello-kokkos-fft hello.cpp)
+target_link_libraries(hello-kokkos-fft PUBLIC Kokkos::kokkos KokkosFFT::fft)
+```
+
+To build the project, we basically rely on the CMake options for Kokkos. For example, the build steps for an A100 GPU based backend are as follows:
+
+```bash
+cmake -B build \
+      -DCMAKE_CXX_COMPILER=g++ \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DKokkos_ENABLE_CUDA=ON \
+      -DKokkos_ARCH_AMPERE80=ON
+cmake --build build -j 8
+```
+
+This way, all the FFT functionalities will be executed on the A100 GPU. For installation as a library, details are provided in the [documentation](https://kokkosfft.readthedocs.io/en/latest/intro/building.html#install-kokkosfft-as-a-library).
```

Comment:
I would drop these two sections and defer to the kokkos-fft documentation.
You could say that the code for the following example can be found in the kokkos-fft repo and post a link.

Reply:
Makes sense. I just tried to make this blog post self-contained. If it is not necessary, I can drop this part